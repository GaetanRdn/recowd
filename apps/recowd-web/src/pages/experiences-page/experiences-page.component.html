<rc-work-in-progress-card></rc-work-in-progress-card>
