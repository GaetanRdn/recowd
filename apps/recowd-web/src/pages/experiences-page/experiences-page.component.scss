:host {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  padding: 2rem 0;

  rc-experience-cards-list {
    margin: auto auto 4rem;
    width: clamp(42.5rem, 60.2rem, 100%);
  }
}
