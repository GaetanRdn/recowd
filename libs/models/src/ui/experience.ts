import { Role } from './role';

export interface Experience {
  client: string;
  localization: string;
  roles: Role[];
  description: string;
  tasks: string[];
  technologies: string[];
}
