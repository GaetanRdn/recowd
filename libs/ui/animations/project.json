{
  "projectType": "library",
  "sourceRoot": "libs/ui/animations/src",
  "prefix": "rc",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:ng-packagr-lite",
      "outputs": [
        "dist/libs/ui/animations"
      ],
      "options": {
        "project": "libs/ui/animations/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/ui/animations/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/ui/animations/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/ui/animations/**/*.ts",
          "libs/ui/animations/**/*.html"
        ]
      }
    }
  },
  "tags": [
    "ui"
  ]
}
