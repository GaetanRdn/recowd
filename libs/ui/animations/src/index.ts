export * from './lib/rotate';
export * from './lib/zoom';
