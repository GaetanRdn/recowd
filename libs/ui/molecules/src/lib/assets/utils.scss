@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
