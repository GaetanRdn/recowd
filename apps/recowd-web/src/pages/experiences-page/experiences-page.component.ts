import { ChangeDetectionStrategy, Component } from '@angular/core';
import { WorkInProgressCardComponent } from '@recowd/ui/molecules';

@Component({
  selector: 'rc-experiences-page',
  standalone: true,
  imports: [WorkInProgressCardComponent],
  templateUrl: './experiences-page.component.html',
  styleUrls: ['./experiences-page.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ExperiencesPageComponent {}
