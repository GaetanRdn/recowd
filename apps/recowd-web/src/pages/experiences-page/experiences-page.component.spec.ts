import { TestBed, waitForAsync } from '@angular/core/testing';

import { ExperiencesPageComponent } from './experiences-page.component';
import { TemplateLookup } from '@recowd/test/utils';
import { TranslateModule } from '@ngx-translate/core';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ExperienceService } from '../../services/experience.service';
import { of } from 'rxjs';

describe('ExperiencesPageComponent', () => {
  let templateLookup: TemplateLookup<ExperiencesPageComponent>;
  let experienceService: ExperienceService;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      imports: [
        ExperiencesPageComponent,
        TranslateModule.forRoot(),
        NoopAnimationsModule,
      ],
      providers: [
        { provide: ExperienceService, useValue: { getAll: jest.fn() } },
      ],
    }).compileComponents();

    experienceService = TestBed.inject(ExperienceService);
    jest.spyOn(experienceService, 'getAll').mockReturnValueOnce(
      of([
        {
          client: 'Vif',
          localization: 'La Chapelle/Erdre, Pays de Loire, France',
          description:
            "Développement from scratch d'une application web à destination des TPEs pour la gestion de leurs produits, clients, commandes et factures.",
          roles: [
            {
              contractType: 'CDI ESN',
              endDate: '2019-06-31',
              startDate: '2017-12-31',
              titre: 'Développeur full-stack',
            },
          ],
          technologies: [
            'Angular 7',
            'Angular Material 7',
            'Java 8',
            'maven',
            'tomcat',
            'Gitlab CI',
          ],
          tasks: [
            'Développement des features',
            "Challenger les maquettes fournies par l'UI/UX",
            'Maintenance corrective et évolutive',
          ],
        },
        {
          client: 'Oui.sncf',
          localization: 'Nantes, Pays de Loire, France',
          description:
            "Maintenance évolutive et corrective sur le back-office de l'application web de devis, vente et après vente de la SNCF.",
          roles: [
            {
              contractType: 'CDI ESN',
              endDate: '2017-12-31',
              startDate: '2014-10-01',
              titre: 'Développeur backend-end',
            },
          ],
          technologies: ['REST', 'Java 8', 'Git', 'Maven', 'Tomcat'],
          tasks: [
            'Développement de features / corrections',
            'Chiffrages',
            'Cadrage technique',
          ],
        },
      ])
    );
    templateLookup = new TemplateLookup(ExperiencesPageComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.fixture.nativeElement.children).toMatchSnapshot());
});
