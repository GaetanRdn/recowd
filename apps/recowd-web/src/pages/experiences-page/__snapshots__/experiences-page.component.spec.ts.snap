// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperiencesPageComponent should create 1`] = `
HTMLCollection [
  <rc-work-in-progress-card>
    <rc-card>
      <div
        class="rc-card-header"
      >
        <rc-card-title
          class="rc-card-title"
        >
          work-in-progress
        </rc-card-title>
      </div>
      <rc-card-content
        class="rc-card-content"
      >
        <rc-work-in-progress-icon>
          <rc-material-icon
            class="material-icons-outlined"
            type="outlined"
          >
            engineering
          </rc-material-icon>
        </rc-work-in-progress-icon>
      </rc-card-content>
    </rc-card>
  </rc-work-in-progress-card>,
]
`;
